### PR TITLE
fix(core/workflow-steps): onStepSelected event 

### PR DESCRIPTION
**Pull request by goncalosard**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Unit tests (`yarn test`) were run locally and passed
- [ ] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [ ] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [x] Bug fix
- [ ] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

[IX-314]

## What is the new behavior?

- Now onStepSelected works correctly emiting the index of the selected step.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
Unit test case missing

**Review by danielleroux — CHANGES_REQUESTED** [review]
Fix not working:

```tsx
        <ix-workflow-steps
          onStepSelected={(event) => {
            console.log(event.detail);
          }}
        >
          <ix-workflow-step>Step 0</ix-workflow-step>
          <ix-workflow-step>Step 1</ix-workflow-step>
          <ix-workflow-step>Step 2</ix-workflow-step>
        </ix-workflow-steps>
```

Should print the index of the step.

**Review by danielleroux — APPROVED** [review]

File: BREAKING_CHANGES.md [hunk 1]
```diff
@@ -2,6 +2,32 @@
 
 ## v2.0.0
 
+### `ix-workflow-steps`
+
+### Remove `linear` prop
+
+Progression of the workflow steps have to be handled by the business logic. We decided this 
+on the base of different user feedbacks.
```
Comment (nuke-ellington) [inline BREAKING_CHANGES.md:10]:
```suggestion
Progression of the workflow steps has to be handled by the business logic now. This change is due to various user feedback.
```

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
Small change - lgtm otherwise

**Review by nuke-ellington — APPROVED** [review]
lgtm

**Pull request merged**